src/tools/compiletest/src/runtest.rs

```diff
@@ -1845,19 +1842,36 @@ impl<'test> TestCx<'test> {
             None => {}
         }
 
+        // Musl toolchain is build on linux-gnu host
+        // but with proper setup it can behave almost* like native linux-musl.
+        // One difference is "cc" which will link to glibc; force musl cc.
         if self.props.force_host {
             self.maybe_add_external_args(&mut rustc,
                                          self.split_maybe_args(&self.config.host_rustcflags));
+            if self.config.target.contains("musl") {
+                if let Some(ref linker) = self.config.linker {
+                    rustc.arg(format!("-Clinker={}", linker));
+                }
+            }
         } else {
             self.maybe_add_external_args(&mut rustc,
                                          self.split_maybe_args(&self.config.target_rustcflags));
             if !is_rustdoc {
                 if let Some(ref linker) = self.config.linker {
                     rustc.arg(format!("-Clinker={}", linker));
                 }
+            } else if self.config.target.contains("musl") {
+                if let Some(ref linker) = self.config.linker {
+                    rustc.arg(format!("--linker={}", linker));
```

Comment:
Is this supposed to be `-C linker`? If so, it looks like this may not be getting executed?

Reply:
It's for `rustdoc` which picks `cc` from the PATH causing build to fail with undefined references to glibc.

Reply:
Er sorry, I didn't realize rustdoc grew a `--linker` option which duplicates `-C linker` ...

Reply:
All fine, it confused me as well when I gave it `-C linker` and got errors about GLIBC symbols :grin:.